src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/event/CaseEventReceiverIT.java

```diff
@@ -0,0 +1,131 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.event;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.ctp.common.FixtureHelper;
+import uk.gov.ons.ctp.common.event.model.CaseEvent;
+import uk.gov.ons.ctp.common.event.model.CaseUpdate;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Case;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.CollectionExercise;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Survey;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.CaseRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.CollectionExerciseRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.PostgresTestBase;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.SurveyRepository;
+
+public class CaseEventReceiverIT extends PostgresTestBase {
+  private static final String CASE_ID = "ad24e36c-2a61-11ec-aa00-4c3275913db5";
+  private static final String SURVEY_ID = "3883af91-0052-4497-9805-3238544fcf8a";
+  private static final String COLLECTION_EX_ID = "4883af91-0052-4497-9805-3238544fcf8a";
+
+  @Autowired private CaseRepository caseRepo;
+  @Autowired private TransactionalOps txOps;
+
+  private CaseEvent caseEvent;
+  private CaseUpdate caseUpdate;
+  private Survey survey;
+
+  @BeforeEach
+  public void setup() {
+    txOps.deleteItems();
+    caseEvent = FixtureHelper.loadPackageFixtures(CaseEvent[].class).get(0);
+    caseUpdate = caseEvent.getPayload().getCaseUpdate();
+    caseUpdate.setCaseId(CASE_ID);
+    caseUpdate.setCollectionExerciseId(COLLECTION_EX_ID);
+    caseUpdate.setSurveyId(SURVEY_ID);
+  }
+
+  @Test
+  public void shouldReceiveCase() {
+    survey = txOps.createSurvey();
+    txOps.createCollex(survey);
```

Comment:
It's a fairly minor point (so won't argue the case if you disagree) but I think it would be good to pass down the critical survey/collex data into their create methods. I'm thinking of things like the survey-id and collex-id. Things that are critical to producing a cohesive test data scenario. 
This would make it more obvious to future readers how the various objects are connected.
There is a bunch of miscellaneous survey/collex fields that are set in the create method. Most of them are unimportant and irrelevant to the test scenario, so am not suggesting that like the surveyName, etc, are exposed.